classes/Entities/Course.php

```diff
@@ -25,16 +25,37 @@ public function courseModules()
         return $this->hasMany(CourseModule::class, 'course', 'id');
     }
 
+    /**
+     * @return \Illuminate\Database\Eloquent\Relations\HasOne
+     */
     public function category()
     {
         return $this->hasOne(CourseCategory::class, 'id', 'category');
     }
 
     /**
-     * @return mixed
+     * @return \Illuminate\Database\Eloquent\Relations\HasMany
      */
     public function courseCohortSyncs()
     {
         return $this->hasMany(CourseCohortSync::class,'courseid','id');
     }
+
+    /**
+     * @return \Illuminate\Database\Eloquent\Relations\HasMany
+     */
+    public function enrolments(){
+        return $this->hasMany(Enrol::class,'courseid','id');
+    }
+
+
+    /**
+     * Get all the enrol id for the user enrolled
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\HasManyThrough
+     */
+    public function userEnrolment(){
+        return $this->hasManyThrough(HelpdeskProduct::class, HelpdeskProductCourse::class, 'coursecategoryid', 'id', 'id', 'id');
```

Comment:
I think this needs to be re-looked at. A function named 'userEnrolment' in the course Model makes me think this gets all users enrolled into the course. But looking at the relationships we are instead linking this to helpdesk, which doesnt really make sense. 

If we want to know what Helpdesk product(s) this course relates to we should instead have a belongsToMany helpdeskProducts() function.